`storage`: use `absl::btree_multimap<>` for `ntp_by_gc_size`

WillemKauf opened the pull request:
### BLUF 

This map is used in the `log_manager` to order `ntp`s by the amount of data they have for reclamation during urgent garbage collection.

Currently, if multiple `ntp`s were to have the same amount of data available for reclaim, the entry in this map for that `size_t` key would only maintain the first `ntp` added, and therefore only one `ntp` of the set would end up being considered for urgent garbage collection.

This is particularly bad since we are only considering the local retention, and not the total available bytes that could be reclaimed due to data being uploaded to tiered storage (i.e a common case here is `usage.reclaim.retention == 0` for multiple `ntp`s, despite the fact that resource management has identified data is available for reclaim from tiered storage enabled topics).

### Demonstrative CI Failure 

[CORE-8500](https://redpandadata.atlassian.net/browse/CORE-8500)

This was manifesting as a CI failure in `full_disk_test.py::LogStorageMaxSizeSI.test_stay_below_target_size` after merging `min.cleanable.dirty.ratio` changes in PR https://github.com/redpanda-data/redpanda/pull/24991.

What this means is that this test was actually depending on regularly scheduled `gc()` every `log_compaction_interval_ms` in addition to urgent `gc()`, and it was because of this bug.

For example, log lines from a failed test:

```
INFO  2025-03-06 04:34:51,766 [shard 0:main] resource_mgmt - storage.cc:656 - Scheduling 12.215MiB for reclaim
TRACE 2025-03-06 04:34:51,766 [shard 1:main] resource_mgmt - storage.cc:356 - Marking offset 223 in cloud partition 1 group {kafka/target-size-topic/1} for estimated 12.215MiB reclaim
INFO  2025-03-06 04:34:51,766 [shard 0:main] resource_mgmt - storage.cc:327 - Requested truncation from 1 cloud partitions
DEBUG 2025-03-06 04:34:51,933 [shard 0:main] storage - log_manager.cc:966 - triggered gc
DEBUG 2025-03-06 04:34:51,933 [shard 0:main] storage - log_manager.cc:432 - Performing urgent gc
DEBUG 2025-03-06 04:35:16,729 [shard 0:main] storage - log_manager.cc:483 - emplacing back usage retention 0 for {redpanda/controller/0}
DEBUG 2025-03-06 04:35:16,730 [shard 0:main] storage - log_manager.cc:483 - emplacing back usage retention 0 for {kafka/target-size-topic/1}
DEBUG 2025-03-06 04:35:16,730 [shard 0:main] storage - log_manager.cc:483 - emplacing back usage retention 0 for {kafka/target-size-topic/3}
DEBUG 2025-03-06 04:35:16,730 [shard 0:main] storage - log_manager.cc:505 - Performing urgent gc on {redpanda/controller/0}
```

Firstly, resource management has identified `target-size-topic/1` as having available bytes for reclaim, and has requested truncation from this partition.

As can be seen, on shard `0`, we have 3 partitions (`controller/0`, `target-size-topic/1`, `target-size-topic/3`) that all have `0` bytes available for removal, according to the local retention policy. Because `controller/0` was considered first, it is the only `ntp` that ultimately has garbage collection performed on it.

In this context, it blocks urgent garbage collection from collecting what it knows to be available bytes from tiered storage topics, and the test fails to reclaim the data it wants to reclaim.

### Change

Fix this by using a `btree_multimap` for the `ntp_by_gc_size` map to prevent this collision case from occuring and blocking urgent garbage collection of certain `ntp`s.


## Backports Required

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

### Bug Fixes

* Fixes a bug in which urgent garbage collection of multiple partitions may only consider one partition in the set.


[CORE-8500]: https://redpandadata.atlassian.net/browse/CORE-8500?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

WillemKauf commented:
/ci-repeat 10
skip-units
dt-repeat=20
tests/rptest/tests/full_disk_test.py::LogStorageMaxSizeSI.test_stay_below_target_size

WillemKauf commented:
/ci-repeat 10
skip-units
skip-redpanda-build
dt-repeat=20
tests/rptest/tests/full_disk_test.py::LogStorageMaxSizeSI.test_stay_below_target_size

WillemKauf commented:
/ci-repeat 10
skip-units
dt-repeat=20
tests/rptest/tests/full_disk_test.py::LogStorageMaxSizeSI.test_stay_below_target_size

WillemKauf commented:
/ci-repeat 10
skip-units
dt-repeat=20
tests/rptest/tests/full_disk_test.py::LogStorageMaxSizeSI.test_stay_below_target_size

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--bb7dd175-34fd-463f-a49a-abb08c3f3f2c-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62687">build#62687</a></summary>


| test_id                                                                                                                     | test_kind   | job_url                                                                                   | test_status   | passed   |
|:----------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568c7-7851-42f9-88f6-0cc5052c198e | FLAKY         | 20/21    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568c7-7851-43e3-89c0-dd97e0955f55 | FLAKY         | 20/21    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568c7-7851-4bcd-a15a-c57aecc27f42 | FLAKY         | 20/22    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568c7-7852-428e-b348-8d894fa329ee | FLAKY         | 20/23    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568c7-7852-449d-b88c-dcb40ce2811d | FLAKY         | 20/22    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568c7-7852-4d4a-aade-b72cd9132e92 | FLAKY         | 20/22    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568c7-7852-4fdb-8bbd-bbb75db9a96e | FLAKY         | 20/23    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568c7-7853-42d6-bdbd-dff42c858319 | FLAKY         | 20/21    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568db-e6a2-42aa-9804-4ad592c02f47 | FLAKY         | 20/23    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568db-e6a3-4382-9bcc-5617db91d318 | FLAKY         | 20/26    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568db-e6a3-4816-a114-6eb58651eb1b | FLAKY         | 20/26    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568db-e6a3-4ae1-926a-f17e14728543 | FLAKY         | 20/27    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568db-e6a3-4dd4-be1c-a36e94147adb | FLAKY         | 20/30    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568db-e6a3-4e2e-9654-18c1ffdfbe9c | FLAKY         | 20/29    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568db-e6a4-400c-94f7-0626ed8b0c6b | FLAKY         | 20/26    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568db-e6a4-4323-8bfb-2d1175ef7a8e | FLAKY         | 20/26    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568db-e6a4-4468-91b7-d4c7eab317c7 | FLAKY         | 20/27    |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62687#019568db-e6a4-4970-9e45-a1d287d4aac0 | FLAKY         | 20/29    |
</details>
<!--/body-->

WillemKauf commented:
/ci-repeat 10
skip-units
dt-repeat=20
tests/rptest/tests/full_disk_test.py::LogStorageMaxSizeSI.test_stay_below_target_size

WillemKauf committed:
`storage`: use `btree_multimap<>` for `ntp_by_gc_size`

Currently, if multiple `ntp`s were to have the same amount of data available
for reclaim, the entry in this map for that `size_t` key would be overwritten,
and only one `ntp` of the set would end up being garbage collected.

This is particularly bad since we are only considering the local retention,
and not the total available bytes that could be reclaimed due to data
being uploaded to tiered storage (i.e a common case here is
`usage.reclaim.retention == 0` for multiple `ntp`s, despite the fact that
resource management has identified data is available for reclaim from
tiered storage enabled topics).

Fix this by using a `btree_multimap` for the `ntp_by_gc_size` map to prevent
this collision case from occuring and blocking urgent garbage collection of
certain `ntp`s.

nvartolomei commented on src/v/storage/log_manager.cc:
```diff
@@ -439,7 +439,8 @@ ss::future<> log_manager::housekeeping_loop() {
              * deleted doing this safely is tricky.
              */
-            absl::btree_map<size_t, model::ntp, std::greater<>> ntp_by_gc_size;
+            absl::btree_map<size_t, chunked_vector<model::ntp>, std::greater<>>
```
How about `absl::btree_multimap`? Fine as it is too.

WillemKauf replied:
Sounds good!

WillemKauf commented:
Force push to:
* change data structure from `absl::map<size_t, chunked_vector<model::ntp>>` to an `absl::btree_multimap<>`.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/62732-->
<!--a13a8f96-4c7d-4987-8710-d9f1e333a180-->
<!--c11e2a2f-8e54-4584-8e21-72e9aaebbf43-->
<!--ee41ef0f-4f48-41c9-9fe1-58148516f9e0-->
<!--title-->
## Retry command for [Build#62732](https://buildkite.com/redpanda/redpanda/builds/62732)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_dropped_column_no_collision@{"catalog_type":"nessie","cloud_storage_type":1,"produce_mode":"proto2","query_engine":"spark"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_dropped_column_no_collision@{"catalog_type":"nessie","cloud_storage_type":1,"produce_mode":"avro","query_engine":"trino"}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--1bf17b1e-4be3-4e81-b030-3ba466ab8cbc-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62732">build#62732</a></summary>


| test_id                                                                                                                                                                                                                                            | test_kind   | job_url                                                                                   | test_status   | passed   |
|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.datalake.custom_partitioning_test.DatalakeCustomPartitioningTest.test_basic.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC                                                                                 | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c3e-9dd3-4329-a2cc-1d1310420cff | FLAKY         | 1/2      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.AVRO.catalog_type=CatalogType.NESSIE      | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c3e-9dd5-4d58-9f65-4543da2db4d4 | FLAKY         | 1/2      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.AVRO.catalog_type=CatalogType.NESSIE      | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c42-784e-4c96-9db2-ed335e23ede2 | FLAKY         | 1/13     |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.AVRO.catalog_type=CatalogType.REST_JDBC   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c42-784c-4e52-bc14-5eccc6037f45 | FLAKY         | 1/19     |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO2.catalog_type=CatalogType.NESSIE    | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c3e-9dd3-4329-a2cc-1d1310420cff | FAIL          | 0/20     |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO2.catalog_type=CatalogType.NESSIE    | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c42-784d-4db6-a6dc-ee0f5c40fc12 | FLAKY         | 1/2      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO2.catalog_type=CatalogType.NESSIE    | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956cc4-41b3-4526-8ec1-5c7b05011a80 | FLAKY         | 1/6      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO2.catalog_type=CatalogType.REST_JDBC | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c3e-9dd1-48f7-b1b2-79b187ea68ad | FLAKY         | 1/2      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO2.catalog_type=CatalogType.REST_JDBC | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c42-784e-4dcf-9734-09622a9232fd | FLAKY         | 1/4      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO2.catalog_type=CatalogType.REST_JDBC | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956cc4-411b-43b8-aef0-9a3d487a021b | FLAKY         | 1/4      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO2.catalog_type=CatalogType.REST_JDBC | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956cc4-41ff-4fb6-9998-a1e68c124780 | FLAKY         | 1/3      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO3.catalog_type=CatalogType.NESSIE    | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c3e-9dd2-492c-9639-a4f4dd1d7b18 | FLAKY         | 1/3      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO3.catalog_type=CatalogType.NESSIE    | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c42-784c-4e52-bc14-5eccc6037f45 | FLAKY         | 1/2      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO3.catalog_type=CatalogType.REST_JDBC | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c3e-9dd5-4d58-9f65-4543da2db4d4 | FLAKY         | 1/2      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO3.catalog_type=CatalogType.REST_JDBC | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c42-784e-4c96-9db2-ed335e23ede2 | FLAKY         | 1/6      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.AVRO.catalog_type=CatalogType.NESSIE      | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c3e-9dd1-48f7-b1b2-79b187ea68ad | FAIL          | 0/20     |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.AVRO.catalog_type=CatalogType.NESSIE      | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c42-784e-4dcf-9734-09622a9232fd | FAIL          | 0/20     |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.AVRO.catalog_type=CatalogType.NESSIE      | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956cc4-411b-43b8-aef0-9a3d487a021b | FLAKY         | 1/2      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.AVRO.catalog_type=CatalogType.NESSIE      | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956cc4-41ff-4fb6-9998-a1e68c124780 | FLAKY         | 1/11     |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.AVRO.catalog_type=CatalogType.REST_JDBC   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c3e-9dd3-4329-a2cc-1d1310420cff | FLAKY         | 1/3      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.AVRO.catalog_type=CatalogType.REST_JDBC   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c42-784d-4db6-a6dc-ee0f5c40fc12 | FLAKY         | 1/5      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.AVRO.catalog_type=CatalogType.REST_JDBC   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956cc4-41b3-4526-8ec1-5c7b05011a80 | FLAKY         | 1/3      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.PROTO2.catalog_type=CatalogType.REST_JDBC | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c3e-9dd2-492c-9639-a4f4dd1d7b18 | FLAKY         | 1/10     |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.PROTO2.catalog_type=CatalogType.REST_JDBC | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c42-784c-4e52-bc14-5eccc6037f45 | FLAKY         | 1/8      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.PROTO3.catalog_type=CatalogType.NESSIE    | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c3e-9dd3-4329-a2cc-1d1310420cff | FLAKY         | 1/5      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.PROTO3.catalog_type=CatalogType.NESSIE    | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c42-784d-4db6-a6dc-ee0f5c40fc12 | FLAKY         | 1/4      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.PROTO3.catalog_type=CatalogType.REST_JDBC | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c3e-9dd1-48f7-b1b2-79b187ea68ad | FLAKY         | 1/2      |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_select_fails.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO2.catalog_type=CatalogType.REST_JDBC | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62732#01956c3e-9dd3-4329-a2cc-1d1310420cff | FLAKY         | 1/2      |
</details>
<!--/body-->

WillemKauf commented:
/ci-repeat 10
skip-units
skip-redpanda-build
dt-repeat=20
tests/rptest/tests/full_disk_test.py::LogStorageMaxSizeSI.test_stay_below_target_size

WillemKauf commented:
/backport v24.3.x

WillemKauf commented:
/backport v24.2.x
